presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileReader.java

```diff
@@ -286,11 +286,15 @@ public void close()
         rowGroupPosition = 0;
         rowGroupRowCount = 0;
         currentChunkRowCount = 0;
-        input.close();
-        if (decompressor != null) {
-            decompressor.destroy();
-        }
 
+        try {
+            input.close();
+        }
+        finally {
+            if (decompressor != null) {
+                decompressor.destroy();
```

Comment:
what about using `Closer` to avoid `finally` and potentially swallowing of `input.close()`'s excpeption?

```
try (Closer closer = Closer.create()) {
    closer.register(input);
    closer.register(decompressor::destroy);
}
```